backend/app/Http/Controllers/UserController.php

```diff
@@ -36,4 +38,22 @@ public function login(Request $request)
 
         return LoginResource::make(array_merge($user->toArray(), ['token' => $token]));
     }
+
+    public function updateInfo(UserUpdateInfoRequest $request)
+    {
+        $userInfo = $request->validated();
+
+        User::find(auth()->user()->id)->update($userInfo);
+
+        return UserResource::make(array_merge(['message' => 'Profile info updated successfully.'], $userInfo));
+    }
+
+    public function updatePassword(UserUpdatePasswordRequest $request)
+    {
+        $userPassword = $request->validated();
+
+        User::find(auth()->user()->id)->update(['password' => $userPassword['new_password']]);
+
+        return UserResource::make(['message' => 'Password updated successfully.']);
+    }
```

Comment:
request, refactor this into a new controller AuthController, then same with comments for `updateInfo`, it should be function name update